Source/ACE.Database/WorldDatabase.cs

```diff
@@ -73,6 +73,113 @@ public uint GetMaxGuidFoundInRange(uint min, uint max)
             }
         }
 
+        private static ConcurrentDictionary<uint, Weenie> level1Scrolls = new ConcurrentDictionary<uint, Weenie>();
+        private static ConcurrentDictionary<uint, Weenie> level2Scrolls = new ConcurrentDictionary<uint, Weenie>();
+        private static ConcurrentDictionary<uint, Weenie> level3Scrolls = new ConcurrentDictionary<uint, Weenie>();
+        private static ConcurrentDictionary<uint, Weenie> level4Scrolls = new ConcurrentDictionary<uint, Weenie>();
+        private static ConcurrentDictionary<uint, Weenie> level5Scrolls = new ConcurrentDictionary<uint, Weenie>();
+        private static ConcurrentDictionary<uint, Weenie> level6Scrolls = new ConcurrentDictionary<uint, Weenie>();
+
+        public Weenie GetSpellWeenie(uint tier, uint spellID)
+        {
+            Weenie val;
+            switch (tier)
+            {
+                case 1:
+                    if(level1Scrolls.TryGetValue(spellID, out val))
+                    {
+                        return val;
+                    }
+                    else
+                    {
+                        var context = new WorldDbContext();
+                        var result = from weenie in context.Weenie
+                                     join did in context.WeeniePropertiesDID on weenie.ClassId equals did.ObjectId
+                                     where weenie.Type == 34 && did.Type == 28 && did.Value == spellID
+                                     select weenie;
+                        level1Scrolls[spellID] = result.FirstOrDefault();
+                        return result.FirstOrDefault();
```

Comment:
Just a few nitpicks.

result should probably be renamed query.

Instead of returning result.FirstOrDefault(), you should return level1Scrolls[spellID] to avoid running the query in duplicate.